src/ethereum_canister_backend/src/lib.rs

```diff
@@ -66,17 +77,22 @@ async fn setup(
     Ok(())
 }
 
-#[update]
+#[query]
 async fn get_block_number() -> Result<Nat> {
-    let head_block_num = HELIOS
-        .with(|client| client.borrow().clone())
-        .ok_or(EthClientError::NotInitialized)?
+    let helios = global_client()?;
+
+    let head_block_num = helios
         .get_block_number()
         .await
-        .map_err(|err| {
-            ic_cdk::println!("Get block number failed: {err}");
-            EthClientError::RpcFailed("get_block_number".into())
-        })?;
+        .map_err(|err| EthClientError::RpcFailed("get_block_number", err.to_string()))?;
 
     Ok(head_block_num.into())
 }
+
+#[update]
```

Comment:
```suggestion
#[query(composite = true)]
```
composite allows for inter canister calls in query
https://docs.rs/ic-cdk/latest/ic_cdk/attr.query.html

Reply:
With this change, the `erc20_balance_of` returns this error:

```
Error: Failed query call.
Caused by: Failed query call.
  The replica returned a replica error: Replica Error: reject code CanisterError, reject message IC0504: Canister bkyz2-fmaaa-aaaaa-qaaaq-cai violated contract: "ic0_call_cycles_add128" cannot be executed in non replicated query mode, error code Some("IC0504")
```